---
title: "Ecogenomique2_CC2"
output: github_document
---

### Effects of Ocean Acidification on Resident and Active Microbial Communities of Stylophora pistillata

## Question scientifique : est-ce que on obtient les mêmes données en utilisant Dada2 phyloseq que MOTHUR en traitement de donnée de bio-informatique?

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
```

```{r}
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra", "devtools")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
library(dada2)
```

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("BiocStyle")
BiocManager::install("Rhtslib")
```

```{r}
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra", "devtools")
install.packages(.cran_packages) 
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
BiocManager::install(.bioc_packages)
# Load packages into session, and print package version
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
```


```{r}
set.seed(100)
miseq_path <- "/home/rstudio/mydatalocal/Ecogeno_CC3_2"
list.files(miseq_path)
```

## Filtration
# Met les reads forward et reverse dans le même ordre
```{r}
fnFs <- sort(list.files(miseq_path,pattern="_1.fastq.gz"))
fnRs <- sort(list.files(miseq_path,pattern="_2.fastq.gz"))
```

# Extrait les noms des échantillons et met au même format.
```{r}
sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)
fnFs <- file.path(miseq_path, fnFs)
fnRs <- file.path(miseq_path, fnRs)
```

# Affiche les 3 premiers éléments de la liste de fnFS
```{r}
fnFs[1:3]
```
# Affiche les 3 premiers éléments de la liste de fnRs
```{r}
fnRs[1:3]
```

# Graphique de la qualité de fnFs
```{}
library(dada2)
library(Rcpp)
```

```{r}
plotQualityProfile(fnFs[1:2])
```

# Graphique de la qualité de fnRs
```{r}
plotQualityProfile(fnRs[1:2])
```

```{r}
filt_path <- file.path(miseq_path, "filtered")
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq.gz"))
```

# Filtration des reads forward et reverse

```{r}
library(dada2)
library(Rcpp)
```

```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) 
head(out)
```

## Enlève les variants de séquence
# Déréplication

```{r}
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames
```

```{r}
library(dada2)
library(phyloseq)
errF <- learnErrors(filtFs, multithread=TRUE)
```

```{r}
errR <- learnErrors(filtRs, multithread=TRUE)
```


```{r}
plotErrors(errF)
plotErrors(errR)
```

```{r}
dadaFs <- dada(derepFs, err=errF, multithread=TRUE)
```

```{r}
dadaRs <- dada(derepRs, err=errR, multithread=TRUE)
```

```{r}
dadaFs[[1]]
```
## Construction de table de séquences et on va enlever les chimères

```{r}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll <- makeSequenceTable(mergers[!grepl("Mock", names(mergers))])
table(nchar(getSequences(seqtabAll)))
```

```{r}
seqtabNoC <- removeBimeraDenovo(seqtabAll)
```

## Assignement taxonomique

```{r}
fastaRef <-"/home/rstudio/silva_nr99_v138.1_train_set.fa.gz"
taxTab<-assignTaxonomy(seqtabNoC, refFasta=fastaRef, multithread=TRUE)
unname(head(taxTab))
```

## Construction de l'arbre phylogénétique

```{r}
library(DECIPHER)
library(BiocGenerics)
library(Biostrings)
seqs <- getSequences(seqtabNoC)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```

```{r}
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
```

```{r}
phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
```

```{r}
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phangAlign)
```

#```{r}
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
        rearrangement = "stochastic", control = pml.control(trace = 0))
detach("package:phangorn", unload=TRUE)
#```

# L'arbre phylogénétique est très lourd et met du temps à charger.

## Combinaison des data dans un objet phyloseq

#```{r}
library(readr)
Meta_donnes<- read.csv(file = "~/mydatalocal/meta_donnes.txt")
Meta_donnes$SampleID <- paste0(gsub("00", "", Meta_donnes$BioSample), "D")
Meta_donnes <- Meta_donnes[!duplicated(Meta_donnes$SampleID),] 
rownames(seqtabAll) <- gsub("124", "125", rownames(seqtabAll))
all(rownames(seqtabAll) %in% Meta_donnes$Run) # TRUE
#```

#```{r}
rownames(Meta_donnes) <- Meta_donnes$SampleID
keep.cols <- c("Run","Assay Type","AvgSpotLen","Bases","BioProject","BioSample","Bytes","Center Name","Consent","DATASTORE filetype","DATASTORE provider","DATASTORE region","ENA-FIRST-PUBLIC (run)","ENA-FIRST-PUBLIC","ENA-LAST-UPDATE (run)","ENA-LAST-UPDATE","Experiment","External_Id","INSDC_center_alias","INSDC_center_name","INSDC_first_public","INSDC_last_update","INSDC_status","Instrument","Library Name","LibraryLayout","LibrarySelection","LibrarySource","Organism","Platform","ReleaseDate","Sample Name","sample_name","SRA Study","Submitter_Id","SampleID") 
Meta_donnes <- Meta_donnes[rownames(seqtabAll), keep.cols],try(ps_ccpna$CCA$eig, silent = TRUE)
#```


#```{r}
library(dada2)
library(phyloseq)
ps <- phyloseq(otu_table(seqtabNoC, taxa_are_rows=FALSE), 
               sample_data(Meta_donnes), 
               tax_table(taxTab),phy_tree(fitGTR$tree))
ps <- prune_samples(sample_names(ps) != "Mock", ps) # Remove mock sample
ps
#```


